### PR TITLE
add more testing around dictionary encoding of traces

### PR DESCRIPTION
**Pull request by richardstartin**

### CODE REVIEW

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import static org.msgpack.core.MessageFormat.INT16
+import static org.msgpack.core.MessageFormat.INT32
+import static org.msgpack.core.MessageFormat.INT64
+import static org.msgpack.core.MessageFormat.INT8
+import static org.msgpack.core.MessageFormat.NEGFIXINT
+import static org.msgpack.core.MessageFormat.POSFIXINT
+import static org.msgpack.core.MessageFormat.UINT16
+import static org.msgpack.core.MessageFormat.UINT32
+import static org.msgpack.core.MessageFormat.UINT64
+import static org.msgpack.core.MessageFormat.UINT8
```
Comment (bantonsson) [inline dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy:30]: `import static org.msgpack.core.MessageFormat.*`?

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy [hunk 2]
```diff
@@ -13,22 +12,30 @@ import org.msgpack.core.MessageUnpacker
...
 import java.nio.ByteBuffer
 import java.nio.channels.WritableByteChannel
-import java.util.concurrent.ThreadLocalRandom
-import java.util.concurrent.TimeUnit
 
+import static datadog.trace.common.writer.ddagent.TraceGenerator.generateRandomTraces
 import static org.junit.Assert.assertEquals
-
-import static org.msgpack.core.MessageFormat.*
+import static org.msgpack.core.MessageFormat.FLOAT32
```
Comment (bantonsson) [inline dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:18]: Why change this?

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+class TraceMapperV04PayloadTest extends DDSpecification {
+
+  def "test dictionary compressed traces written correctly" () {
```
Comment (bantonsson) [inline dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy:34]: Wrong test name?

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy [hunk 4]
```diff
@@ -39,15 +46,42 @@ class TraceMapperV05PayloadTest extends DDSpecification {
     verifier.verifyTracesConsumed()
 
     where:
-    bufferSize    |   traceCount
-    100 << 10     |       0
-    100 << 10     |       1
-    100 << 10     |       10
-    100 << 10     |       100
+    bufferSize    | dictionarySize |   traceCount   | lowCardinality
+    10 << 10      |   10 << 10     |       0        | true
```
Comment (bantonsson) [inline dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:50]: Why the shifts here? I don't get them.

Comment (richardstartin) [reply dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:50]: `<< 10` turns the number on the left into kilobytes

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy (hunk 3 above)
Comment (richardstartin) [reply dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV04PayloadTest.groovy:34]: yes

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy (hunk 2 above)
Comment (richardstartin) [reply dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:18]: automatic change

File: dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy (hunk 4 above)
Comment (bantonsson) [reply dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:39]: So this is your fancy way of writing `* 1024`. I see.

Comment (richardstartin) [reply dd-trace-core/src/test/groovy/datadog/trace/common/writer/ddagent/TraceMapperV05PayloadTest.groovy:39]: Well, if you want to write MB it's `<< 20` which is less of a mouthful than `* 1024 * 1024`

**Pull request merged**